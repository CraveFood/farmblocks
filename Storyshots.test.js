import ReactDOM from "react-dom";
import initStoryshots, {
  multiSnapshotWithOptions,
} from "@storybook/addon-storyshots";
import styleSheetSerializer from "jest-styled-components/src/styleSheetSerializer";
import { addSerializer } from "jest-specific-snapshot";

// add jest-styled-components serializer in order for jest-styled-components to work
// see https://github.com/storybooks/storybook/issues/887#issuecomment-356577444
// and https://github.com/storybooks/storybook/issues/887#issuecomment-357073807
addSerializer(styleSheetSerializer);

jest.spyOn(ReactDOM, "createPortal").mockImplementation(node => node);

initStoryshots({
  framework: "react",
  test: multiSnapshotWithOptions({}),
});
