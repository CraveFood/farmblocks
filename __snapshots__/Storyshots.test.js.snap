// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Label Default 1`] = `
.c1 {
  font-family: Lato,sans-serif;
  font-size: 14px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #2f313a;
  text-align: left;
  line-height: 16px;
  font-weight: 600;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  text-transform: uppercase;
}

.c0 {
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  margin-bottom: 8px;
  float: left;
}

<div>
  <div
    className="labelText c0 c1"
    fontWeight="title"
    letterSpacing="2px"
    size={14}
    type="#2f313a"
  >
    Default Label
  </div>
</div>
`;

exports[`Storyshots Label Disabled 1`] = `
.c1 {
  font-family: Lato,sans-serif;
  font-size: 14px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: rgba(0,0,0,0.32);
  text-align: left;
  line-height: 16px;
  font-weight: 600;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  text-transform: uppercase;
}

.c0 {
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  margin-bottom: 8px;
  float: left;
}

<div>
  <div
    className="labelText c0 c1"
    fontWeight="title"
    letterSpacing="2px"
    size={14}
    type="rgba(0, 0, 0, 0.32)"
  >
    Disabled Label
  </div>
</div>
`;

exports[`Storyshots Label Focused 1`] = `
.c1 {
  font-family: Lato,sans-serif;
  font-size: 14px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #3498db;
  text-align: left;
  line-height: 16px;
  font-weight: 600;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  text-transform: uppercase;
}

.c0 {
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  margin-bottom: 8px;
  float: left;
}

<div>
  <div
    className="labelText c0 c1"
    fontWeight="title"
    letterSpacing="2px"
    size={14}
    type="#3498db"
  >
    Focused Label
  </div>
</div>
`;

exports[`Storyshots Label Invalid 1`] = `
.c1 {
  font-family: Lato,sans-serif;
  font-size: 14px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #dd3311;
  text-align: left;
  line-height: 16px;
  font-weight: 600;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  text-transform: uppercase;
}

.c0 {
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  margin-bottom: 8px;
  float: left;
}

<div>
  <div
    className="labelText c0 c1"
    fontWeight="title"
    letterSpacing="2px"
    size={14}
    type="#dd3311"
  >
    Invalid Label
  </div>
</div>
`;

exports[`Storyshots Label Protected and disabled 1`] = `
.c1 {
  font-family: Lato,sans-serif;
  font-size: 14px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #2f313a;
  text-align: left;
  line-height: 16px;
  font-weight: 600;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  text-transform: uppercase;
}

.c0 {
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  margin-bottom: 8px;
  float: left;
}

<div>
  <div
    className="labelText c0 c1"
    fontWeight="title"
    letterSpacing="2px"
    size={14}
    type="#2f313a"
  >
    Protected Label
  </div>
</div>
`;

exports[`Storyshots Label With more info 1`] = `
.c5 {
  position: relative;
  display: inline-block;
}

.c5 .hit-area {
  position: absolute;
  top: 0;
  width: 100%;
  height: 100%;
  cursor: pointer;
}

.c5 .appear-enter {
  opacity: 0;
}

.c5 .appear-enter-active {
  opacity: 1;
  -webkit-transition: opacity 200ms ease-in;
  transition: opacity 200ms ease-in;
}

.c5 .appear-exit {
  opacity: 1;
}

.c5 .appear-exit-active {
  opacity: 0;
  -webkit-transition: opacity 200ms ease-out;
  transition: opacity 200ms ease-out;
}

.c4 {
  font-family: Lato,sans-serif;
  font-size: 16px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #59636f;
  text-align: left;
  font-weight: 600;
}

.c1 {
  font-family: Lato,sans-serif;
  font-size: 14px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #2f313a;
  text-align: left;
  line-height: 16px;
  font-weight: 600;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  text-transform: uppercase;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.c3 .tooltip {
  margin-left: -4px;
}

.c3 .icon {
  display: inline-block;
  margin: 0 8px;
  cursor: pointer;
  color: #59636f;
}

.c3 .icon:hover,
.c3 .hovered {
  color: #3498db;
}

.c0 {
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  margin-bottom: 8px;
  float: left;
}

.c2 {
  position: relative;
  top: -1px;
}

<div>
  <div
    className="labelText c0 c1"
    fontWeight="title"
    letterSpacing="2px"
    size={14}
    type="#2f313a"
  >
    Default Label
  </div>
  <div
    className="moreInfo c2 c3"
  >
    <div
      className="text  c4"
      fontWeight="title"
      size={16}
      type="#59636f"
    />
    <div
      className="icon c5"
    >
      <div
        className="hit-area"
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
      />
      <i
        className="wg-question"
      />
    </div>
  </div>
</div>
`;

exports[`Storyshots Label With more info right aligned 1`] = `
.c5 {
  position: relative;
  display: inline-block;
}

.c5 .hit-area {
  position: absolute;
  top: 0;
  width: 100%;
  height: 100%;
  cursor: pointer;
}

.c5 .appear-enter {
  opacity: 0;
}

.c5 .appear-enter-active {
  opacity: 1;
  -webkit-transition: opacity 200ms ease-in;
  transition: opacity 200ms ease-in;
}

.c5 .appear-exit {
  opacity: 1;
}

.c5 .appear-exit-active {
  opacity: 0;
  -webkit-transition: opacity 200ms ease-out;
  transition: opacity 200ms ease-out;
}

.c4 {
  font-family: Lato,sans-serif;
  font-size: 16px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #59636f;
  text-align: left;
  font-weight: 600;
}

.c1 {
  font-family: Lato,sans-serif;
  font-size: 14px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #2f313a;
  text-align: left;
  line-height: 16px;
  font-weight: 600;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  text-transform: uppercase;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.c3 .tooltip {
  margin-right: -4px;
}

.c3 .icon {
  display: inline-block;
  margin: 0 8px;
  cursor: pointer;
  color: #59636f;
}

.c3 .icon:hover,
.c3 .hovered {
  color: #3498db;
}

.c0 {
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  margin-bottom: 8px;
  float: left;
}

.c2 {
  position: relative;
  top: -1px;
}

<div>
  <div
    className="labelText c0 c1"
    fontWeight="title"
    letterSpacing="2px"
    size={14}
    type="#2f313a"
  >
    Default Label
  </div>
  <div
    className="moreInfo c2 c3"
  >
    <div
      className="text  c4"
      fontWeight="title"
      size={16}
      type="#59636f"
    />
    <div
      className="icon c5"
    >
      <div
        className="hit-area"
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
      />
      <i
        className="wg-question"
      />
    </div>
  </div>
</div>
`;

exports[`Storyshots Label extended style 1`] = `
.c2 {
  font-family: Lato,sans-serif;
  font-size: 14px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #2f313a;
  text-align: left;
  line-height: 16px;
  font-weight: 600;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  text-transform: uppercase;
}

.c1 {
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  margin-bottom: 8px;
  float: left;
}

.c0 {
  border: dotted 4px sienna;
}

.c0::after {
  display: block;
  content: '';
  clear: both;
}

<div
  className="c0"
>
  <div
    className="labelText c1 c2"
    fontWeight="title"
    letterSpacing="2px"
    size={14}
    type="#2f313a"
  >
    Default Label
  </div>
</div>
`;
